zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractPlugin.java

```diff
@@ -616,14 +618,17 @@ protected boolean isPage200(HttpMessage msg) {
     }
 
     /**
-     * Tells whether or not the message matches {@code CustomPage.Type.ERROR_500} definitions.
+     * Tells whether or not the message matches {@code CustomPage.Type.ERROR_500} definitions. Falls
+     * back to simply checking the response status code for "500 - Internal Server Error".
      *
      * @param msg the message that will be checked
      * @return {@code true} if the message matches, {@code false} otherwise
      * @since TODO Add version
      */
     protected boolean isPage500(HttpMessage msg) {
-        return isCustomPage(msg, CustomPage.Type.ERROR_500);
+        boolean is500 =
+                msg.getResponseHeader().getStatusCode() == HttpStatusCode.INTERNAL_SERVER_ERROR;
+        return is500 ? is500 : isCustomPage(msg, CustomPage.Type.ERROR_500);
```

Comment:
For 404 and 200 types would it make more sense to check analyser first? (Performance wise it seemed lighter here to just check the status first.)